flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DefaultSlotPoolFactory.java

```diff
@@ -75,10 +82,21 @@ public static DefaultSlotPoolFactory fromConfiguration(
 		final Time rpcTimeout = AkkaUtils.getTimeoutAsTime(configuration);
 		final Time slotIdleTimeout = Time.milliseconds(configuration.getLong(JobManagerOptions.SLOT_IDLE_TIMEOUT));
 
+		final SchedulingStrategy schedulingStrategy = selectSchedulingStrategy(configuration);
+
 		return new DefaultSlotPoolFactory(
 			rpcService,
+			schedulingStrategy,
 			SystemClock.getInstance(),
 			rpcTimeout,
 			slotIdleTimeout);
 	}
+
+	private static SchedulingStrategy selectSchedulingStrategy(Configuration configuration) {
+		if (configuration.getBoolean(CheckpointingOptions.LOCAL_RECOVERY)) {
+			return PreviousAllocationSchedulingStrategy.getInstance();
+		} else {
+			return LocationPreferenceSchedulingStrategy.getInstance();
```

Comment:
Does it make sense to use the `LocationPreferenceSchedulingStrategy` for the init scheduling even when the "local recovery" is enable?

Reply:
It is used if the `Execution` don't have a previous location assigned. You can see this in the implementation of the `PreviousAllocationSchedulingStrategy` which extends `LocationPreferenceSchedulingStrategy`.